tests/components/mikrotik/test_device_tracker.py

```diff
@@ -0,0 +1,118 @@
+            hass,
+            device_tracker.DOMAIN,
+            {device_tracker.DOMAIN: {"platform": "mikrotik"}},
+        )
+        is False
+    )
+    assert mikrotik.DOMAIN not in hass.data
+
+
+async def test_device_trackers(hass):
+    """Test device_trackers created by mikrotik."""
+
+    # test devices are added from wireless list only
+    hub = await setup_mikrotik_entry(hass)
+
+    device_1 = hass.states.get("device_tracker.device_1")
+    assert device_1 is not None
+    assert device_1.state == "home"
+    device_2 = hass.states.get("device_tracker.device_2")
+    assert device_2 is None
+
+    with patch.object(mikrotik.hub.MikrotikData, "command", new=mock_command):
+        # test device_2 is added after connecting to wireless network
+        WIRELESS_DATA.append(DEVICE_2_WIRELESS)
+
+        await hub.async_update()
+        await hass.async_block_till_done()
+
+        device_2 = hass.states.get("device_tracker.device_2")
+        assert device_2 is not None
+        assert device_2.state == "home"
+
+        # test state remains home if last_seen  consider_home_interval
+        del WIRELESS_DATA[1]  # device 2 is removed from wireless list
+        hub.api.devices["00:00:00:00:00:02"]._last_seen = dt_util.utcnow() - timedelta(
+            minutes=4
+        )
+        await hub.async_update()
+        await hass.async_block_till_done()
+
+        device_2 = hass.states.get("device_tracker.device_2")
+        assert device_2.state != "not_home"
+
+        # test state changes to away if last_seen > consider_home_interval
+        hub.api.devices["00:00:00:00:00:02"]._last_seen = dt_util.utcnow() - timedelta(
+            minutes=5
+        )
+        await hub.async_update()
+        await hass.async_block_till_done()
+
+        device_2 = hass.states.get("device_tracker.device_2")
+        assert device_2.state == "not_home"
+
+
+async def test_restoring_devices(hass):
+    """Test restoring existing device_tracker entities if not detected on startup."""
+    config_entry = MockConfigEntry(
+        domain=mikrotik.DOMAIN, data=MOCK_DATA, options=MOCK_OPTIONS
+    )
+    config_entry.add_to_hass(hass)
+
+    registry = await entity_registry.async_get_registry(hass)
+    registry.async_get_or_create(
+        device_tracker.DOMAIN,
+        mikrotik.DOMAIN,
+        "00:00:00:00:00:01",
+        suggested_object_id="device_1",
+        config_entry=config_entry,
+    )
+    registry.async_get_or_create(
+        device_tracker.DOMAIN,
+        mikrotik.DOMAIN,
+        "00:00:00:00:00:02",
+        suggested_object_id="device_2",
+        config_entry=config_entry,
+    )
+
+    await setup_mikrotik_entry(hass)
+
+    # test device_2 which is not in wireless list is restored
+    device_1 = hass.states.get("device_tracker.device_1")
+    assert device_1 is not None
+    assert device_1.state == "home"
+    device_2 = hass.states.get("device_tracker.device_2")
+    assert device_2 is not None
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4918093)